src/SessionSetup/ConnectToStadiaWidget.cpp

```diff
@@ -264,12 +264,21 @@ void ConnectToStadiaWidget::ReloadInstances() {
   });
 }
 
-void ConnectToStadiaWidget::CheckCredentialsAvailable() {
+void ConnectToStadiaWidget::CheckCredentialsAvailableOrLoad() {
   CHECK(selected_instance_.has_value());
 
   const std::string instance_id = selected_instance_->id.toStdString();
 
-  if (!instance_credentials_.contains(instance_id)) return;
+  if (!instance_credentials_.contains(instance_id)) {
+    if (!instance_credentials_loading_.contains(instance_id)) {
+      ggp_client_->GetSshInfoAsync(
+          selected_instance_.value(),
+          [this, instance_id = instance_id](outcome::result<orbit_ggp::SshInfo> ssh_info_result) {
+            OnSshInfoLoaded(std::move(ssh_info_result), instance_id);
+          });
```

Comment:
There is a `instance_credentials_loading_.emplace(instance_id)` missing.

Reply:
Indeed, that doesn't make a lot of sense without it... I wonder where that went because I remember writing this line at some point. Thanks for catching this, fixed!